trainer/finetune.py

```diff
@@ -0,0 +1,240 @@
+# Copyright 2020 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Script for Running, Training, and Evaluating E2E Convrec Experiments."""
+
+# SETUP
+print("Installing dependencies...")
+import functools
+import os
+import time
+import warnings
+import tensorflow.compat.v1 as tf
+import tensorflow_datasets as tfds
+import t5
+from contextlib import contextmanager
+import logging as py_logging
+import gzip
+import json
+import os
+import tensorboard as tb
+import random
+import nltk
+import sacrebleu
+from absl import app
+from absl import flags
+from trainer import preprocessing
+
+FLAGS = flags.FLAGS
+flags.DEFINE_integer('steps', 6000, "Finetuning training steps.")
+flags.DEFINE_enum("size", "base", ["small", "base", "large", "3B", "11B"], "model size")
+flags.DEFINE_string("name", "default", "name/description of model  version")
+flags.DEFINE_enum("mode", "all", ["train", "evaluate", "all"], "run mode: train, evaluate, or all")
+
+INPUT_LENGTH = 512 #1033 - longest training input for redial
+TARGET_LENGTH = 128 #159 - longest trainin target for redial
+
+warnings.filterwarnings("ignore", category=DeprecationWarning)
```

Comment:
Is there a lot of deprecated stuff in our dependencies here? :)

Reply:
I think a lot of stuff in tensorflow v1 gives depreciation warnings. I just kept this because it was in the notebook, but I could look into it in more detail to see what exactly we're using that is depreciated.